rspace/src/main/scala/coop/rchain/rspace/state/RSpaceExporter.scala

```diff
@@ -19,115 +19,89 @@ object RSpaceExporter {
 
   final case object EmptyHistoryException extends Exception
 
-  type ReadParams[F[_]] = (
-      // Keys (tries) to read
-      Vector[TrieNode[Blake2b256Hash]],
-      // Offset path
-      Seq[(Blake2b256Hash, Option[Byte])],
-      // Skip & take counter
-      Counter,
-      // Result tries as indexed collection
-      Vector[TrieNode[Blake2b256Hash]]
-  )
-
-  def traverseTrie[F[_]: Monad](
+  def traverseHistory[F[_]: Sync](
       startPath: Seq[(Blake2b256Hash, Option[Byte])],
       skip: Int,
       take: Int,
-      getTrie: Blake2b256Hash => F[Trie]
+      getFromHistory: ByteVector => F[Option[ByteVector]]
   ): F[Vector[TrieNode[Blake2b256Hash]]] =
-    startPath.headOption.fold(Vector.empty[TrieNode[Blake2b256Hash]].pure[F]) {
-      case (root, _) =>
-        val startParams: ReadParams[F] = (
-          // First node is root / start path head
-          Vector(TrieNode(root, isLeaf = false, Seq.empty)),
-          startPath,
-          Counter(skip, take),
-          Vector.empty
+    if (startPath.isEmpty) Vector[TrieNode[Blake2b256Hash]]().pure
+    else {
+      import coop.rchain.rspace.history.RadixTree._
+      import scodec.bits.ByteVector
```

Comment:
This is duplicated import already defined on top.

Please add empty line after imports for better clarity.